core/src/main/scala/caliban/rendering/DocumentRenderer.scala

```diff
@@ -327,7 +327,8 @@ object DocumentRenderer extends Renderer[Document] {
       definition match {
         case SchemaDefinition(directives, query, mutation, subscription, description) =>
           val hasTypes    = query.nonEmpty || mutation.nonEmpty || subscription.nonEmpty
-          val isExtension = directives.nonEmpty && !hasTypes
+          val isExtension =
+            (!hasTypes && directives.nonEmpty) || query.isEmpty && (mutation.nonEmpty || subscription.nonEmpty)
```

Comment:
Based on your comment, shouldn't this be:

```suggestion
            (!hasTypes && directives.nonEmpty) || (query.isEmpty && (mutation.nonEmpty || subscription.nonEmpty))
```

Reply:
whoops, I thought I had done it and the formatter stripped it, but I must have made a mistake. Fixed